src/backend/resourcemanager/requesthandler.c

```diff
@@ -627,7 +627,11 @@ bool handleRMSEGRequestIMAlive(void **arg)
 	fts_client_ip_len = strlen(fts_client_ip);
 	inet_aton(fts_client_ip, &fts_client_addr);
 	fts_client_host = gethostbyaddr(&fts_client_addr, 4, AF_INET);
-	Assert(fts_client_host != NULL);
+	if (fts_client_host == NULL)
+	{
+		elog(WARNING, "Failed to reverse DNS lookup for ip %s.", fts_client_ip);
+		return true;
+	}
 
```

Comment:
Not quit after "reverse dns lookup" failure. This function handles the IMAlive message from segment resource manager process.  "return true" just ignore this message and do not continue the followup work after a lookup failure.  IP address is not enough since followup work needs the hostname.

For replace gethostbyaddr(), @jiny2 could you give some comments ?